src/WebUI/Pods/PodsRightPanel.tsx

```diff
@@ -85,12 +87,25 @@ export class PodsRightPanel extends React.Component<IPodRightPanelProps, IPodsRi
             selectedTab: selectedPivot,
             selectedImageDetails: undefined,
             showImageDetails: (imageId: string) => {
-                const imageService = KubeFactory.getImageService();
-                imageService && imageService.getImageDetails(imageId).then(imageDetails => {
+                const showImageDetails = (imageDetails: IImageDetails): void => {
                     this.setState({
                         selectedImageDetails: imageDetails
                     });
-                });
+                }
+
+                let imageDetails: IImageDetails | undefined = StoreManager.GetStore<ImageDetailsStore>(ImageDetailsStore).getImageDetails(imageId);
```

Comment:
would it be good to have autil function in action creator or in some util file which can take an imageid and an action and then dot he following?

let imageDetails: IImageDetails | undefined = StoreManager.GetStore<ImageDetailsStore>(ImageDetailsStore).getImageDetails(imageId);
                if (imageDetails) {
                    showImageDetails(imageDetails);
                }
                else {
                    const imageService = KubeFactory.getImageService();
                    imageService && imageService.getImageDetails(imageId).then(imageDetails => {
                        if (imageDetails) {
                            ActionsCreatorManager.GetActionCreator<ImageDetailsActionsCreator>(ImageDetailsActionsCreator).setImageDetails(imageDetails);
                            showImageDetails(imageDetails);
                        }
                    });
                }
            }	            }


as this code is duplicated?

Reply:
Done